plugins/azure/resoto_plugin_azure/resource/compute.py

```diff
@@ -3231,6 +3233,10 @@ class AzureVirtualMachineSize(AzureResource, BaseInstanceType):
     number_of_cores: Optional[int] = field(default=None, metadata={'description': 'The number of cores supported by the virtual machine size. For constrained vcpu capable vm sizes, this number represents the total vcpus of quota that the vm uses. For accurate vcpu count, please refer to https://docs. Microsoft. Com/azure/virtual-machines/constrained-vcpu or https://docs. Microsoft. Com/rest/api/compute/resourceskus/list.'})  # fmt: skip
     os_disk_size_in_mb: Optional[int] = field(default=None, metadata={'description': 'The os disk size, in mb, allowed by the virtual machine size.'})  # fmt: skip
     resource_disk_size_in_mb: Optional[int] = field(default=None, metadata={'description': 'The resource disk size, in mb, allowed by the virtual machine size.'})  # fmt: skip
+    location: Optional[str] = field(default=None, metadata={"description": "Resource location."})
+
+    def pre_process(self, graph_builder: GraphBuilder, source: Json) -> None:
+        self.location = graph_builder.location.name if graph_builder.location else ""
```

Comment:
All location-dependent resources use the same pattern: they are loaded via a GraphBuilder with location.
Please see GraphBuilder.add_node - there is already handling for defining the region in this case. Maybe this is already sufficient?

Reply:
We need to know location to connect `AzureVirtualMachineSize` between `AzureVirtualMachine`. VirtualMachineSize by its own does not have location value.

And in GraphBuilder.add_node `add_node` method does not apply location on resource for further comparison.

Reply:
It does. After you added the vmSize to the graph, the location is available via `region`.
No objection to introduce it as separate property.